### PR TITLE
add default error boundry to project

### PR DESCRIPTION
**Pull request by benschac**
## What does this PR do and why?
What: This adds an error boundry to the entire project. It's basically a giant try catch block.
Why: If there's a piece of client side JS that throws (and isn't handled) that will crash the app with a white screen.

 We want a way to tell the user that something went wrong. Additionally, we want to log it in Sentry. 

Mostly copy-pasta from the nextjs docs

How to test: Add a `throw Error('test error')` in the project, the application will show a prompt asking the user to reload the page. I feel like we should design something a bit nicer. I thought this was okay for the moment. 

There isn't much in the project for error screens at the moment. This could be a good opportunity to design 500, 400, client side errors, and project not found pages. 

CleanShot 2022-12-05 at 11 01 14@2x

More here: https://nextjs.org/docs/advanced-features/error-handling#handling-client-errors
_Provide a description of what this PR does. Link to any relevant GitHub issues, Notion tasks or Discord discussions._

## Screenshots or screen recordings
<>
_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by benschac** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2616** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2616" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2616?utm_source=stack-comment).

File: src/components/common/ErrorBoundry.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        client_side_exception: errorInfo.componentStack,
+      },
+    })
+
+    console.error('Uncaught error:', error, errorInfo)
+  }
+
+  public render() {
+    if (this.state.hasError) {
+      return <h1>Sorry.. there was an error</h1>
```
Comment (benschac) [inline src/components/common/ErrorBoundry.tsx:34]: Trans tags aren't instantiated yet. Assuming just having English copy is alright for the moment. 


The other thought I had was keeping this Error boundary as a last case error catch and adding another error boundary that's inside the i18n provider.

**Comment by benschac** [comment]
additionally, I think there's more to be desired in terms of a error page. This could be a good time to have standard 400, 500, and project not found, and client side error pages built across the project

**Comment by benschac** [comment]
@wraeth-eth -- would you mind taking a look at this PR while @tomquirk is away? He LGTM'ed the bundle analyzer PR. I don't want to assume this PR has an LGTM, would love some extra 👀

**Comment by tomquirk** [comment]
Hey @benschac thanks for this, although i think we can opt for a Next-idiomatic approach here: https://nextjs.org/docs/advanced-features/custom-error-page

**Pull request closed without merging**